spec/cucumber/formatter/html_spec.rb

```diff
@@ -33,7 +33,7 @@ module Formatter
       describe 'when writing the report to a file' do
         before(:each) do
           allow(@out).to receive(:respond_to?).with(:path).and_return(true)
-          expect(@out).to receive(:path).and_return('out/file.html')
+          expect(@out).to receive(:path).at_most(:twice).and_return('out/file.html')
```

Comment:
I'd be more comfortable if you set the limit within the scope of where it will be encountered, rather than setting it at a level that potentially affects every test. In other words, set it to 1 in

```
describe 'with a step that embeds a snapshot' do
  # stuff
end
```

and set it to 2 in

```
describe 'with more than one step that embeds snapshot' do
  #stuff
end
```

Everything else looks fine.

Reply:
Updated. I had to move calling `:run_defined_feature` method into a nested _describe_ sections because `:path` should be stubbed before using `:run_defined_feature`.